### ISSUE TITLE
[2017.7] Two tests failing on Fedora 26

### ISSUE THREAD

**Issue by rallytime** [saltstack/salt-jenkins#477]
https://jenkins.saltstack.com/job/branch_tests/job/2017.7/job/salt-linode-fedora26/lastCompletedBuild/testReport/

- integration.shell.test_auth.AuthTest.test_pam_auth_valid_user
- integration.shell.test_auth.AuthTest.test_zzzz_tearDown

**Comment by terminalmage** [comment]
The `zzzz` failure was due to the failure of the other test. It looks like that test was actually failing because of a different bug (which appears to have been fixed), having to do with the output of the call being made. The branch build for F26 2017.7 is now passing.

I've moved the weird tearDown test to an actual tearDown in https://github.com/saltstack/salt/pull/42885, and will keep an eye on this branch build to see if it continues to pass.

**Comment by rallytime** [comment]
@terminalmage It appears that the first test (test_pam_auth_valid_user) is still failing on Fedora 26: https://jenkins.saltstack.com/job/branch_tests/job/2017.7/job/salt-linode-fedora26/30/

**Comment by terminalmage** [comment]
Thanks, I'll take a look.

**Comment by terminalmage** [comment]
I can't get this to fail alone in a Docker container, so I've dropped some debug logging in and am currently doing a full test suite run.

**Comment by terminalmage** [comment]
I wasn't able to reproduce this in a full run in Docker, either.

**Comment by terminalmage** [comment]
I've added some debug logging to troubleshoot this further in https://github.com/saltstack/salt/pull/42925

**Comment by terminalmage** [comment]
OK, after adding the debug logging, I have found that the auth is failing. The output of the test.ping command is:

`Failed to authenticate! This is most likely because this user is not permitted to execute commands, but there is a small possibility that a disk error occurred (check disk/inode usage).`

This test runs `shadow.set_password` to set the password before running the test.ping, but `shadow.set_password` runs with no logging for obvious security reasons, so there's no logging present to help us on this. I'll keep digging.

**Comment by terminalmage** [comment]
I've found the issue. The setUp is trying to create a user called `saltdev` but it is failing to do so because there is already a group present by the name `saltdev`. So, some other test in the suite is creating the `saltdev` group and not successfully cleaning it afterward, and since /etc/login.defs has `USERGROUPS_ENAB yes`, useradd tries to create the group as part of the user creation process, and when it can't it just doesn't add the user. Since the user does not exist, the PAM auth (rightly) fails.

Trying to track down where this isn't being cleaned up.

**Comment by terminalmage** [comment]
OK, I was slightly mistaken in my earlier comment. It turns out that this is being caused by sssd, a caching daemon in the vein of nscd. It turns out that sssd is utter trash, and can't seem to deal with users and groups being created in short succession with the same uid/gid. I tried several methods to run different `sss_cache` commands, both before and after creating and deleting the conflicting users/groups, and the tests continued to fail. So, after wasting far more time screwing around with this than it deserved, I've decided to just disable sssd via #491.
